src/frontend/Komponenter/Personoversikt/Infotrygdperioderoversikt.tsx

```diff
@@ -70,17 +71,99 @@ const SummertePerioder: React.FC<{ perioder: SummertPeriode[] }> = ({ perioder }
     );
 };
 
+type InfotrygdPeriodeMedFlereEndringer = InfotrygdPeriode & {
+    initialKode?: Kode;
+};
+
+/**
+ * Då det finnes flere endringer på en periode/vedtak så har noen av de høyere presedense en andre
+ * Vanligt er att man har Førstegangsvedtak/Endring/G-omregning, og sen får ett opphør, då skal opphøret vises primært
+ */
+const mapKode = (kode: Kode): number => {
+    switch (kode) {
+        case Kode.OVERTFØRT_NY_LØSNING:
+            return 5;
+        case Kode.UAKTUELL:
+            return 4;
+        case Kode.OPPHØRT:
+            return 3;
+        default:
+            return 0;
+    }
+};
+
+const formatStønadTom = (periode: InfotrygdPeriodeMedFlereEndringer): string => {
+    const stønadTom = formaterNullableMånedÅr(periode.stønadTom) as string;
+    if (periode.kode === Kode.OPPHØRT) {
+        return `${formaterNullableMånedÅr(periode.opphørsdato)} (${stønadTom})`;
+    } else {
+        return stønadTom;
+    }
+};
+
+const grupperPerioderPerVedtak = (
+    perioder: InfotrygdPeriode[]
+): { [key: string]: InfotrygdPeriode[] } =>
+    perioder.reduce((acc, periode) => {
+        const prev = acc[periode.vedtakId] || [];
+        prev.push(periode);
+        acc[periode.vedtakId] = prev;
+        return acc;
+    }, {} as { [key: string]: InfotrygdPeriode[] });
+
+/**
+ * På ett vedtak kan det finnes flere endringer (1-2), de slås sammen, men beholder initielle koden, for å kunne vise i tabellen
+ */
+const slåSammenVedtak = (
+    perioderPerVedtak: InfotrygdPeriode[][]
+): InfotrygdPeriodeMedFlereEndringer[] =>
+    perioderPerVedtak.reduce((acc: InfotrygdPeriodeMedFlereEndringer[], perioder) => {
+        if (perioder.length === 1) {
+            acc.push(perioder[0]);
+        } else {
+            const sortertePerioder = perioder.sort((a, b) =>
+                mapKode(a.kode) > mapKode(b.kode) ? -1 : 1
+            );
+            acc.push({
+                ...sortertePerioder[0],
+                initialKode: sortertePerioder[1].kode,
```

Comment:
Her tar vi ut koden til periode #2 i lista - er det ønskelig? Basert på kommentaren skulle man trodd at initiell enten er `0` eller `length-1` ?

Reply:
De blir sortert med siste endringen først, og då den tidligere endringen i `1`

Reply:
Men det kan bare være to? Eller kan det bli flere? Hvis flere så bør vel `length-1` brukes istedenfor `1` ?

Reply:
Det finnes 2 tilfeller med 3, disse har jeg bare ignorert 👼  har lagt til kommentar om det nå 👍  bra catch